app/src/main/java/openfoodfacts/github/scrachx/openfood/repositories/ProductRepository.java

```diff
@@ -134,9 +139,12 @@ private ProductRepository() {
      */
     @Override
     public Single<List<Label>> getLabels(Boolean refresh) {
-        if (refresh || tableIsEmpty(labelDao)) {
+        Long lastModifiedDate = UpdateSinceLastUpload("labels");
+        if (lastModifiedDate > 1) {
```

Comment:
Ideal: replace Long by a simple class containing an Enum ( ERROR, UPTODATE, TO_DOWNLOAD and SKIPPED) and the timestamp.
It's more explicit